js/ui/diagram/ui.diagram.js

```diff
@@ -49,11 +50,81 @@ class Diagram extends Widget {
         this._createComponent($options, DiagramOptions);
 
         !isServerSide && this._diagramInstance.createDocument($content[0]);
+        this._diagramInstance.onChanged = this._raiseValueChangeAction.bind(this);
     }
     _initDiagram() {
         const { DiagramControl } = getDiagram();
         this._diagramInstance = new DiagramControl();
     }
+
+    _getDefaultOptions() {
+        return extend(super._getDefaultOptions(), {
+            /**
+            * @name dxDiagramOptions.onValueChanged
+            * @extends Action
+            * @type function(e)
+            * @type_function_param1 e:object
+            * @type_function_param1_field4 value:object
+            * @type_function_param1_field5 previousValue:object
+            * @action
+            */
+            onValueChanged: null,
```

Comment:
Should it have "on**Value**Changed" action? The `value` option related to our Editors. Editors has a validation, readOnly state and other features that our users can expect from the widget with the `value` option.

Perhaps, it should be something like `onStateChanged` to avoid misunderstandings?

Reply:
It's not state, it's more like onDiagramChanged, onDocumentChanged, or just onChanged.
But we have no options Diagram or Document. It will be still inconsistent with your approaches.
And, if I understand correctly your code, instance.on("somethingChanged", ...) will work only if we have the "something" option, right?
Is there any way out here?